app/assets/javascripts/discourse/app/components/emoji-picker.js

```diff
@@ -245,15 +255,152 @@ export default Component.extend({
 
   @action
   keydown(event) {
+    const arrowKeys = ["ArrowDown", "ArrowUp", "ArrowLeft", "ArrowRight"];
+    const recentEmojis = document.querySelectorAll(this.elements.recentEmojis);
+    const searchBar = document.querySelector(this.elements.searchBar);
+    let emojis = document.querySelectorAll(this.elements.emojiResults);
+    const firstSection = document.querySelectorAll(
+      this.elements.firstEmojiSection
+    );
+
+    let currentEmoji;
+
+    this.set(
+      "hoveredEmoji",
+      this._codeWithDiversity(event.target.title, this.selectedDiversity)
+    );
+
+    // if no search results get all emojis
+    if (emojis.length === 0) {
+      emojis = document.querySelectorAll(this.elements.allEmojis);
+    }
+
+    if (
+      event.code === "ArrowDown" &&
+      this._focusedOn(this.elements.searchBar)
+    ) {
+      return emojis[0].focus();
+    }
+
     if (event.code === "Escape") {
       this.onClose(event);
       return false;
     }
+
+    if (arrowKeys.includes(event.code)) {
+      if (!this._focusedOn(this.elements.picker)) {
+        return;
+      }
+
+      Array.from(emojis).find((e, index) => {
+        currentEmoji = index;
+        return e.isEqualNode(event.target);
+      });
+
+      const numEmojisInRow = this.get("emojiPerRow");
+
+      if (event.code === "ArrowRight") {
+        let nextEmoji = currentEmoji + 1;
+
+        if (nextEmoji < emojis.length) {
+          emojis[nextEmoji].focus();
+        } else if (nextEmoji >= emojis.length) {
+          nextEmoji = 0;
+          emojis[nextEmoji].focus();
+        }
+      }
+
+      if (event.code === "ArrowLeft") {
+        const previousEmoji = currentEmoji - 1;
+        if (currentEmoji > 0) {
+          emojis[previousEmoji].focus();
+        }
+      }
+
+      if (event.code === "ArrowDown") {
+        const emojiNextRow = currentEmoji + numEmojisInRow;
+
+        if (
+          event.target.classList.contains("recent-emoji") &&
+          recentEmojis.length < 11
```

Comment:
This `11` here is a little too much magic and I don't think it'll work for mobile (we show 7 per row on mobile AFAIK). Is there a way to avoid this?